enterprise/course_catalog_api.py

```diff
@@ -22,50 +24,126 @@
+            list: List of catalogs available for the user.
+        """
+        return self._load_data('catalogs')
 
-def get_all_catalogs(user):
-    """
-    Return a list of all course catalogs, including name and ID.
-    """
-    client = get_catalog_api_client(user)
-    if CatalogIntegration is None:
-        raise NotConnectedToOpenEdX(
-            _('To get a CatalogIntegration object, this package must be installed in an OpenEdX environment.')
-        )
-    if get_edx_api_data is None:
-        raise NotConnectedToOpenEdX(
-            _('To parse a catalog API response, this package must be installed in an OpenEdX environment.')
+    def get_course_run(self, course_run_id):
+        """
+        Return course_run data, including name, ID and seats.
+
+        Args:
+            course_run_id(str): Course run ID (aka Course Key) in string format.
+
+        Returns:
+            dict: Course run data provided by Course Catalog API.
+        """
+        return self._load_data('course_runs', resource_id=course_run_id)
+
+    def get_program_by_title(self, program_title):
+        """
+        Return single program by name, or None if not found.
+
+        Arguments:
+            program_title(str): Program title as seen by students and in Course Catalog Admin
+
+        Returns:
+            dict: Program data provided by Course Catalog API
+        """
+        all_programs = self._load_data('programs')
+        matching_programs = [program for program in all_programs if program.get('title') == program_title]
+        if len(matching_programs) > 1:
+            raise MultipleProgramMatch(len(matching_programs))
+        elif len(matching_programs) == 1:
+            return matching_programs[0]
+        else:
+            return None
+
+    def get_program_by_uuid(self, program_uuid):
+        """
+        Return single program by UUID, or None if not found.
+
+        Arguments:
+            program_uuid(string): Program UUID in string form
+
+        Returns:
+            dict: Program data provided by Course Catalog API
+        """
+        return self._load_data('programs', resource_id=program_uuid)
+
+    def get_common_course_modes(self, course_run_ids):
+        """
+        Find common course modes for a set of course runs.
+
+        Arguments:
+            course_run_ids(Iterable[str]): Target Course run IDs.
+
+        Returns:
+            set: course modes found in all given course runs
+        """
+        available_course_modes = None
+        for course_run_id in course_run_ids:
+            course_run = self.get_course_run(course_run_id) or {}
+            course_run_modes = {seat.get('type') for seat in course_run.get('seats', [])}
+
+            if available_course_modes is None:
+                available_course_modes = course_run_modes
+            else:
+                available_course_modes &= course_run_modes
+
+            if not available_course_modes:
+                return available_course_modes
```

Comment:
@mattdrayer in this scenario, since the first two does not have run modes the right result would be an empty set. I'll add a test for this case.